lib/validate.js

```diff
@@ -81,6 +120,29 @@ module.exports = function validateMiddleware (opts) {
               msg: 'Bad Request - ' + err.toString()
             });
           } else {
+            // Assemble object from properties deemed valid from
+            // all Joi schema validations.
+            var validParams = _.assign.apply(_, joiedParams);
+
+            // See if we removed any params, if so, log it.
+            var removedParams = _.difference(_.keys(data), _.keys(validParams));
+            if(removedParams.length > 0) {
+              log.info(
+                '%s params removed for "%s" during joi validation: %s. %s %s',
+                (type === 'list') ? 'req.query' : 'req.body',
+                type,
+                removedParams,
+                req.method,
+                req.originalUrl
+              );
+            }
+
+            // List operates on req.query, update and create operate on req.body
+            if(type === 'list') {
```

Comment:
Perhaps we should be nice and make the original params still be available? Something like:

```js
req.fhRest.originalQuery = req.query;
req.query = validParams;
```

Then we could also pass the original query along to the adapter.